src/local/Matcher.ts

```diff
@@ -0,0 +1,62 @@
+interface Data<T = undefined> {
+  criteria: string | RegExp;
+  data: T;
+}
+
+export interface Matcher {
+  add<T>(criteria: string, data?: T): void;
+  match<T = any>(criteria: string): T[];
+  remove<T>(criteria: string, data?: T): void;
+  isWildcardChannel(str: string): boolean;
+}
+
+export default class LocalMatcher implements Matcher {
```

Comment:
What does `LocalMatcher` mean? I don't think it makes sense.

_If_ it should've been renamed, the file name should have been renamed as well. But I don't think this one should have been renamed.

You could use the same name for the interface and the class in this case. The interface would then be implicitly implemented.